guides/hosting/gitlab.md

```diff
@@ -0,0 +1,157 @@
+
+In the _Badges_ section of the _General_ settings, we'll first add a release badge. The link is: `https://gitlab.com/<YOUR-GITLAB-USERNAME>/<YOUR-REPOSITORY-NAME>/pipelines` and the _Badge Image URL_ is: `https://gitlab.com/<YOUR-GITLAB-USERNAME>/<YOUR-REPOSITORY-NAME>/badges/master/pipeline.svg`.
+
+And now if the pipleline has finished we'll have docs and we can link to them with a generic badge from `shields.io`.
+
+- Link: `https://<YOUR-GITLAB-USERNAME>.gitlab.io/<YOUR-REPOSITORY-NAME>`
+- Image: `https://img.shields.io/badge/docs-available-brightgreen.svg`
+
+
+## Releases
+
+A release is just a special commit in your history with a name (see [tagging](https://git-scm.com/book/en/v2/Git-Basics-Tagging)).
+
+According to [the Crystal Shards README](https://gitlab.com/crystal-lang/shards/blob/master/README.md),
+
+> When libraries are installed from Git repositories, the repository is expected to have version tags following a semver-like format, prefixed with a `v`. Examples: v1.2.3, v2.0.0-rc1 or v2017.04.1
+
+GitLab also has a [releases feature](https://docs.gitlab.com/ee/workflow/releases.html) that let's you associate files and a description with this tag. That way you can (for example) distribute binaries.
+
+As you'll see from the [releases docs](https://docs.gitlab.com/ee/workflow/releases.html), you can either create an _annotated_ tag along with release notes/files in the UI:
+
+![gitlab new tags UI](./gitlab_tags_new.png)
+
+or you can create the tag from the command line like so:
+
+```bash
+    $ git tag -a v0.1.0 -m "Release v0.1.0"
+```
+
+push it up
+
+```bash
+    $ git push public master --follow-tags
+```
+
+and then use the UI to add/edit the release note and attach files.
+
+**Best Practices**
+
+* Use the `-a` option to create an annotated tag for releases.
+
+* Follow [Semantic Versioning](http://semver.org/) and create a new release every time you push new code to `master`.
+
+### Release Badge
+
+If you'd like you can also add a `shields.io` badge for the release. GitLab doesn't have full support for this kind of thing, and until someone adds a [version badge for gitlab](https://github.com/badges/shields/blob/master/doc/TUTORIAL.md) to shields.io, we'll have to just code in the version number in the URLs directly.
+
+- Link: `https://img.shields.io/badge/release-<VERSION>-brightgreen.svg`
+- Image: `https://img.shields.io/badge/release-<VERSION>-brightgreen.svg`
+
+where `<VERSION>` is the version number prefixed with a `v` like this: `v0.1.0`.
+
+### Mirror to GitLab
+
+At the moment, crystalshards.xyz only uses the GitHub API, so if you want your library to be indexed on that service you can either:
+
+* Start a discussion about it over on their [github repo](https://github.com/zamith/crystalshards/issues) and eventually submit a Pull Request.
```

Comment:
This is the wrong repo for crystalshards.xyz, and encouraging people to open an issue in an official tutorial is unacceptable, you should start an issue yourself and link to it if anything.